application/src/main/webapp/js/chat.js

```diff
@@ -1648,17 +1659,41 @@ ChatApplication.prototype.getUsers = function(roomId, callback, asString) {
  * @param filter : the filter (ex: Ben Pa)
  * @param callback : return the json users data list as a parameter of the callback function
  */
-ChatApplication.prototype.getAllUsers = function(filter, callback) {
+ChatApplication.prototype.getAllUsers = function(filter, isSearchInRoomFirst, callback) {
+
+  // Cache result
+  var _this = this;
+  if (this.getAllUsersCache == undefined) {
+    this.getAllUsersCache = {};
+  }
+  var cacheKey = filter;
+  if (isSearchInRoomFirst) {
+    cacheKey += '__' + this.targetUser;
+  }
+  if (this.getAllUsersCache[cacheKey] != undefined) {
```

Comment:
Having to refresh the page to see the last data is not what I expect from a Single Page Application. My chat is always open in a browser tab and I never refresh it.
That said, I agree the cases I mentionned are corner cases, so we can go like that for now, but I think we should study the use of a more advanced client cache (home-made or ext lib). WDYT ?